packages/react-components/react-carousel-preview/library/src/utils/useMutationObserver.ts

```diff
@@ -0,0 +1,42 @@
+import type { MutableRefObject } from 'react';
+import * as React from 'react';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+
+const { useRef, useEffect } = React;
+
+export const useMutationObserver = (
+  target: Element | Document | undefined,
+  callback: MutationCallback,
+  options?: MutationObserverInit,
+): {
+  observer: MutableRefObject<MutationObserver | undefined>; // eslint-disable-line no-restricted-globals
+} => {
+  'use no memo';
+
+  // TODO: exclude types from this lint rule: https://github.com/microsoft/fluentui/issues/31286
+  // eslint-disable-next-line no-restricted-globals
+  const observer = useRef<MutationObserver>();
+  const { targetDocument } = useFluent();
+  const win = targetDocument?.defaultView;
+
+  useEffect(() => {
+    if (!win) {
+      return;
+    }
+    // Create an observer instance linked to the callback function
+    observer.current = new win.MutationObserver(callback);
+  }, [callback, win]);
+
+  useEffect(() => {
+    if (target) {
+      // Start observing the target node for configured mutations
+      observer.current?.observe(target, options);
+    }
+
+    return () => {
+      observer.current?.disconnect();
+    };
+  }, [target, options]);
```

Comment:
⚠️ Heads up!

This is very unsafe: `options` are a dependency there:

<img width="569" alt="image" src="https://github.com/microsoft/fluentui/assets/14183168/ba003c61-90b6-476b-b56c-c401552746ec">

At the same time, when the hook is used - they are defined inside `useIntersectionObserver` thus they will create a new reference ➡️ will connect/disconnect on every re-render.

As a workaround, you can do:

```ts
const ATTRS = {
    attributes: true,
    subtree: true,
    attributeFilter: ['dir'],
}

  // Mutation observer for dir attribute changes in the document
  useMutationObserver(targetDocument, mutationObserverCallback);
```

However, as I see `useMutationObserver()` is used once for a single purpose. I suggest to rename it to `useDirAttrObserver()` (or your pick!) and move `options` inside, so it will be just:

```
useDirAttrObserver(mutationObserverCallback)
```

**At the same time**, we can get proper `dir` from `useFluent()` and consumers must configure it anyway, so instead of the observer you can just use `useEffect()` with a dependency on it.